Packs/TrendMicroVisionOne/Integrations/TrendMicroVisionOne/TrendMicroVisionOne.py

```diff
@@ -0,0 +1,1054 @@
+        exception_count = len(list_of_exception)
+        return exception_count
+
+    def get_workbench_histories(self, start, end, offset=None, size=None) -> str:
+        if not check_datetime_aware(start):
+            start = start.astimezone()
+        if not check_datetime_aware(end):
+            end = end.astimezone()
+        start = start.astimezone(timezone.utc)
+        end = end.astimezone(timezone.utc)
+        start = start.isoformat(timespec='milliseconds').replace('+00:00', 'Z')
+        end = end.isoformat(timespec='milliseconds').replace('+00:00', 'Z')
+
+        params = dict([('startTime', start),
+                       ('endTime', end),
+                       ('sortBy', 'createdTime')]
+                      + ([('offset', offset)] if offset is not None else [])
+                      + ([('limit', size)] if size is not None else []))
+
+        response = self.http_request(GET, WORKBENCH_HISTORIES, params=params)['data']['workbenchRecords']
+        return response
+
+
+def test_module(client: Client) -> Any:
+    """
+    Performs basic get request to get item samples.
+    :type client: ``Client``
+    :param client: client object to use http_request.
+    """
+    client.http_request('GET', '/v2.0/xdr/threatintel/suspiciousObjects/exceptions')
+    return 'ok'
+
+
+def add_block_list_mapping(task_status: str, data: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Mapping add to block list response data.
+    :type task_status: ``str``
+    :param task_status: The status of task either success or failed.
+    :type data: ``dict``
+    :param data: Response data to received from the end point.
+    :return: mapped response data.
+    :rtype: ``dict``
+    """
+    action_id = data.get('actionId', {})
+    return {
+        'action_id': action_id,
+        'task_status': task_status
+    }
+
+
+def remove_block_list_mapping(task_status: str, data: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Mapping remove to block list response data.
+    :type task_status: ``str``
+    :param task_status: The status of task either success or failed.
+
+    :type data: ``dict``
+    :param method: Response data to received from the end point.
+    :return: mapped response data.
+    :rtype: ``dict``
+    """
+    action_id = data.get('actionId', {})
+    return {
+        'action_id': action_id,
+        'task_status': task_status
+    }
+
+
+def fetch_incidents(client: Client):
+    """
+    This function do the loop to get all workbench alerts by changing
+    the parameters of both 'offset' and 'size'.
+    """
+    offset = 0
+    size = demisto.params().get('max_fetch')
+    end = datetime.now(timezone.utc)
+    days = int(demisto.params().get('first_fetch'))
+
+    last_run = demisto.getLastRun()
+    if last_run and 'start_time' in last_run:
+        start = datetime.fromisoformat(last_run.get('start_time'))
+    else:
+        start = end + timedelta(days=-days)
+
+    alerts: List[Any] = []
+    while True:
```

Comment:
Please fetch only max_fetch incidents. Currently you use the max_fetch arg to fetch x incidents each http request. Given x as max_fetch arg, please fetch only x incidents. if more incidents available fetch them in the next run.

Reply:
Updated so we only retrieve the maximum in max_fetch.